src/main/java/org/opensearch/security/privileges/SystemIndexAccessEvaluator.java

```diff
@@ -304,7 +304,7 @@ private void evaluateSystemIndicesAccess(
 
         // cluster actions will return true for requestedResolved.isLocalAll()
         // the following section should only be run for index actions
-        if (user.isPluginUser() && !requestedResolved.isLocalAll()) {
```

Comment:
Should we wait for the backport of : https://github.com/opensearch-project/security/pull/5032 to handle this and avoid merge conflicts?

Reply:
There is small overlap between https://github.com/opensearch-project/security/pull/5032 and this PR. I made sure to account for the `plugins.security.system_indices.enabled = false` scenario in this PR.

If there are conflicts I will open up a manual backport for https://github.com/opensearch-project/security/pull/5032